packages/vaadin-grid/src/array-data-provider.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * Returns a sub-property of an object
+ *
+ * @param {string} path dot-separated path to the sub property
+ * @param {*} object
+ * @returns {*}
+ */
+function get(path, object) {
+  return path.split('.').reduce((obj, property) => obj[property], object);
+}
+
+/**
+ * Check array of filters/sorters for paths validity, console.warn invalid items
+ * @param {!Array<!GridFilter | !GridSorter>} arrayToCheck The array of filters/sorters to check
+ * @param {string} action The name of action to include in warning (filtering, sorting)
+ * @param {!Array<!GridItem>} items
+ */
+function checkPaths(arrayToCheck, action, items) {
+  if (!items.length) {
```

Comment:
**nit:** If the `items` argument is accidentally an object, the current check successfully passes and the exception (the function doesn't support objects) is thrown somewhere later so that can make the debugging a bit harder.

```suggestion
  if (items.length === 0) {
```